From: Maxim Devaev <mdevaev@gmail.com>
Date: Fri, 28 Aug 2026 18:30:23 +0300
Subject: [PATCH 21/25] pikvm: gadget: hid: Remove string IDs


PiKVM's 1003-pikvm-gadget-hid-Remove-string-IDs.patch (pikvm/packages, packages/linux-rpi-pikvm),
unchanged. It applies to 6.18 as it is.

---
 drivers/usb/gadget/function/f_hid.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/usb/gadget/function/f_hid.c b/drivers/usb/gadget/function/f_hid.c
index 450fc84..9f254ca 100644
--- a/drivers/usb/gadget/function/f_hid.c
+++ b/drivers/usb/gadget/function/f_hid.c
@@ -283,6 +283,7 @@ static struct usb_descriptor_header *hidg_fs_descriptors_ssreport[] = {
 /*-------------------------------------------------------------------------*/
 /*                                 Strings                                 */
 
+#if 0
 #define CT_FUNC_HID_IDX	0
 
 static struct usb_string ct_func_string_defs[] = {
@@ -299,6 +300,7 @@ static struct usb_gadget_strings *ct_func_strings[] = {
 	&ct_func_string_table,
 	NULL,
 };
+#endif
 
 /*-------------------------------------------------------------------------*/
 /*                              Char Device                                */
@@ -1167,7 +1169,9 @@ static int hidg_bind(struct usb_configuration *c, struct usb_function *f)
 {
 	struct usb_ep		*ep;
 	struct f_hidg		*hidg = func_to_hidg(f);
+#	if 0
 	struct usb_string	*us;
+#	endif
 	int			status;
 
 	hidg->get_req = usb_ep_alloc_request(c->cdev->gadget->ep0, GFP_ATOMIC);
@@ -1179,12 +1183,14 @@ static int hidg_bind(struct usb_configuration *c, struct usb_function *f)
 	hidg->get_req->context = hidg;
 	hidg->get_report_returned = true;
 
+#	if 0
 	/* maybe allocate device-global string IDs, and patch descriptors */
 	us = usb_gstrings_attach(c->cdev, ct_func_strings,
 				 ARRAY_SIZE(ct_func_string_defs));
 	if (IS_ERR(us))
 		return PTR_ERR(us);
 	hidg_interface_desc.iInterface = us[CT_FUNC_HID_IDX].id;
+#	endif
 
 	/* allocate instance-specific interface IDs, and patch descriptors */
 	status = usb_interface_id(c, f);
